### ISSUE TITLE
Coordinate transformation between geocentric cartesian coordinates and geographic coordinates

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#243]
Using ellipsoids defined in #242 add functionality to transform between:
* geographic (geodetic) coordinates (latitude, longitude)
* ECEF (earth-centered earth-fixed) geocentric cartesian coordinates (x, y, z)

**Comment by navispatial** [comment]
Ported from https://github.com/chrisveness/geodesy/blob/master/latlon-ellipsoidal.js by Chris Veness (MIT License).

Ported Dart code for conversions between **geographic coordinates** (latitude and longitude as geodetic coordinates) and  **geocentric cartesian coordinates** represented by ECEF (earth-centric earth-fixed) position:

```dart
Position toGeocentricCartesian() {
    // source geographic position
    final lat = origin.lat.toRadians();
    final lon = origin.lon.toRadians();
    final h = origin.elev;
    final sinLat = sin(lat);
    final cosLat = cos(lat);
    final sinLon = sin(lon);
    final cosLon = cos(lon);

    // ellipsoidal parameters
    final a = ellipsoid.a;
    final f = ellipsoid.f;

    // 1st eccentricity squared ≡ (a²-b²)/a²
    final eSq = 2.0 * f - f * f;

    // (ν = nu) radius of curvature in prime vertical
    final nu = a / sqrt(1.0 - eSq * sinLat * sinLat);

    // target geocentric cartesian position
    return Position.create(
      x: (nu + h) * cosLat * cosLon,
      y: (nu + h) * cosLat * sinLon,
      z: (nu * (1.0 - eSq) + h) * sinLat,
    );
  }

static Geographic fromGeocentricCartesian(
    Position geocentric, {
    Ellipsoid ellipsoid = Ellipsoid.WGS84,
  }) {
    // ε = epsilon, β = beta, ν = nu

    // source geocentric cartesian position
    final x = geocentric.x;
    final y = geocentric.y;
    final z = geocentric.z;

    // ellipsoidal parameters
    final a = ellipsoid.a;
    final b = ellipsoid.b;
    final f = ellipsoid.f;

    final eSq = 2.0 * f - f * f; // 1st eccentricity squared ≡ (a²−b²)/a²
    final epsilon2 = eSq / (1.0 - eSq); // 2nd eccentricity squared ≡ (a²−b²)/b²
    final p = sqrt(x * x + y * y); // distance from minor axis
    final R = sqrt(p * p + z * z); // polar radius

    // parametric latitude (Bowring eqn.17, replacing tanβ = z·a / p·b)
    final tanBeta = (b * z) / (a * p) * (1.0 + epsilon2 * b / R);
    final sinBeta = tanBeta / sqrt(1.0 + tanBeta * tanBeta);
    final cosBeta = sinBeta / tanBeta;

    // geodetic latitude (Bowring eqn.18: tanφ = z+ε²⋅b⋅sin³β / p−e²⋅cos³β)
    final lat = cosBeta.isNaN
        ? 0.0
        : atan2(
            z + epsilon2 * b * sinBeta * sinBeta * sinBeta,
            p - eSq * a * cosBeta * cosBeta * cosBeta,
          );

    // geodetic longitude
    final lon = atan2(y, x);

    // height above ellipsoid (Bowring eqn.7)
    final sinLat = sin(lat);
    final cosLat = cos(lat);
    final nu = a /
        sqrt(
          1.0 - eSq * sinLat * sinLat,
        ); // length of the normal terminated by the minor axis
    final h = p * cosLat + z * sinLat - (a * a / nu);

    // create a geographic position
    return Geographic(
      lat: lat.toDegrees(),
      lon: lon.toDegrees(),
      elev: h,
    );
  }
```

**Comment by navispatial** [comment]
Implemented in [geobase version 1.4.0](https://pub.dev/packages/geobase/versions/1.4.0+1), see [milestone](https://github.com/navibyte/geospatial/milestone/6).
